### PR TITLE
[Merged by Bors] - feat: Port Data.Rat.Sqrt

### PR DESCRIPTION
**Pull request by zaxioms0**
Hi! This is my first port so I picked something pretty small. This is the initial commit where everything compiles properly, but I haven't looked at style guidelines super thoroughly yet. Please give any feedback that you have!

### CODE REVIEW

**Review by Ruben-VandeVelde — APPROVED** [review]
Looks good as far as I can tell without CI. Please ask for push access to the main repository on zulip, so you can make branches there going forward.

**Comment by xoers** [comment]
will do!

File: Mathlib/Data/Rat/Sqrt.lean [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+/-- Square root function on rational numbers, defined by taking the (integer) square root of the
+numerator and the square root (on natural numbers) of the denominator. -/
+-- @[pp_nodot] porting note: unknown attribute
+def sqrt (q : ℚ) : ℚ := mkRat (Int.sqrt q.num) (Nat.sqrt q.den)
+#align rat.sqrt Rat.sqrt
+
+theorem sqrt_eq (q : ℚ) : Rat.sqrt (q * q) = |q| := by
+  rw [sqrt, mul_self_num, mul_self_den, Int.sqrt_eq, Nat.sqrt_eq, abs_def, divInt_ofNat]
+
```
Comment (ChrisHughes24) [inline Mathlib/Data/Rat/Sqrt.lean:34]:
```suggestion
```

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**